GoogleForCreators/web-stories-wp
# New Story Details Modal: Add Experiment

## Context

Initial wiring of new story publish modal via experiment. 

## Summary

So as to not do the rest of this feature in a vacuum - when  `enableUpdatedPublishStoryModal` is true and you click 'publish' you get the new (unfinished) modal instead of the existing modal for the checklist. 


## Relevant Technical Choices



## To-do

(See https://github.com/GoogleForCreators/web-stories-wp/issues/10115 for full list)

## User-facing changes

N/A

## Testing Instructions

- See that publishing works the same on this branch as on staging until you turn on the aforementioned experiment (`enableUpdatedPublishStoryModal`).
- With the experiment enabled see that when you click "publish" in the editor you get the new modal. You can update the story title and description here and then continue with the "publish" button in the modal and it'll publish like normal. 

<img width="1261" alt="Screen Shot 2022-02-04 at 12 27 41 PM" src="https://user-images.githubusercontent.com/10720454/152593983-9a9cad65-762d-42f4-a137-9608f5611da7.png">




This PR can be tested by following these steps:

1.


## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Partially addresses #10115
